Sources/Extensions/Alamofire/AlamofireRequest+Extensions.swift

```diff
@@ -22,121 +22,43 @@
 import RxAlamofire
 
-typealias ServerResponse = (response: HTTPURLResponse, result: Any)
+typealias ServerResponse = (HTTPURLResponse, Data)
 
 public extension Reactive where Base: DataRequest {
 
-    private typealias JSON = [String: Any]
-
     /// Method that serializes response into target object
     ///
     /// - Parameter mappingQueue: The dispatch queue to use for mapping
+    /// - Parameter decoder: JSONDecoder used to decode a decodable type
     /// - Returns: Observable with HTTP URL Response and target object
-    func apiResponse<T: ImmutableMappable>(mappingQueue: DispatchQueue = .global())
+    func apiResponse<T: Decodable>(mappingQueue: DispatchQueue = .global(), decoder: JSONDecoder)
         -> Observable<(response: HTTPURLResponse, model: T)> {
 
-        return responseJSONOnQueue(mappingQueue)
-            .tryMapResult { resp, value in
-                let json = try cast(value) as JSON
-
-                return (resp, try T(JSON: json))
-            }
-    }
-
-    /// Method that serializes response into array of target objects
-    ///
-    /// - Parameter mappingQueue: The dispatch queue to use for mapping
-    /// - Returns: Observable with HTTP URL Response and array of target objects
-    func apiResponse<T: ImmutableMappable>(mappingQueue: DispatchQueue = .global())
-        -> Observable<(response: HTTPURLResponse, models: [T])> {
-
-            return responseJSONOnQueue(mappingQueue)
-                .tryMapResult { resp, value in
-                    let jsonArray = try cast(value) as [JSON]
-
-                    return (resp, try Mapper<T>().mapArray(JSONArray: jsonArray))
-                }
-    }
-
-    /// Method that serializes response into target type
-    ///
-    /// - Parameter mappingQueue: The dispatch queue to use for mapping
-    /// - Returns: Observable with HTTP URL Response and target object
-    func apiResponse<T>(mappingQueue: DispatchQueue = .global())
-        -> Observable<(response: HTTPURLResponse, object: T)> {
-
-            return responseJSONOnQueue(mappingQueue)
-                .tryMapResult { resp, value in
-                    (resp, try cast(value) as T)
+            return responseData()
+                .observeOn(SerialDispatchQueueScheduler(queue: mappingQueue, internalSerialQueueName: mappingQueue.label))
+                .tryMapResult { response, data in
+                    (response, try decoder.decode(T.self, from: data))
                 }
     }
 
     /// Method that serializes response into target object
     ///
     /// - Parameter mappingQueue: The dispatch queue to use for mapping
     /// - Returns: Observable with HTTP URL Response and target object
-    func observableApiResponse<T: ObservableMappable>(mappingQueue: DispatchQueue = .global())
+    func observableApiResponse<T: ObservableMappable>(mappingQueue: DispatchQueue = .global(), decoder: JSONDecoder)
         -> Observable<(response: HTTPURLResponse, model: T)> where T.ModelType == T {
```

Comment:
update `ObservableMappable` and remove generic constraints in functions

```swift
public protocol ObservableMappable where ModelType == Self {

```

Reply:
do you even need ModelType then? Maybe
```swift
public protocol ObservableMappable where Self: Decodable {

    static func create(from jsonObject: Any, with decoder: JSONDecoder) -> Observable<Self>
}

```

Reply:
maybe. try it)